ynr/apps/candidates/models/popolo_extra.py

```diff
@@ -191,6 +194,31 @@ def get_bulk_add_review_url(self):
     def get_sopn_url(self):
         return self.get_absolute_url("ballot_paper_sopn")
 
+    def get_results_url(self):
+        """
+        Get url to direct user to enter results. Currently only FPTP
+        supports full results, otherwise we direct user to the ballot
+        view where they can mark individual candidates as elected.
+        """
+        if self.is_fptp:
+            return self.get_absolute_url(viewname="ballot_paper_results_form")
+        return self.get_absolute_url()
+
+    @property
+    def results_button_text(self):
+        """
+        Allows different button text depending on the voting system
+        """
+        mapping = {self.VOTING_SYSTEM_FPTP: "Add results"}
+        return mapping.get(self.voting_system, "Mark elected candidates")
+
+    @property
+    def is_fptp(self):
```

Comment:
Can we change this logic to not encode the voting type, but to workout if results entering is supported? e.g `can_enter_votes_cast` would do `voting_system in ["FPTP", "other supported type"]`